rmw_fastrtps_shared_cpp/src/participant.cpp

```diff
@@ -161,19 +163,78 @@ rmw_fastrtps_shared_cpp::create_participant(
   eprosima::fastdds::dds::DomainParticipantQos domainParticipantQos =
     eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->get_default_participant_qos();
 
+  // Configure discovery
+  switch (discovery_options->automatic_discovery_range) {
+    case RMW_AUTOMATIC_DISCOVERY_RANGE_NOT_SET:
+      RMW_SET_ERROR_MSG("automatic discovery range must be set");
+      return nullptr;
+      break;
+    case RMW_AUTOMATIC_DISCOVERY_RANGE_OFF:
+      domainParticipantQos.wire_protocol().builtin.discovery_config.discoveryProtocol =
+        eprosima::fastrtps::rtps::DiscoveryProtocol_t::NONE;
```

Comment:
This will prevent readers and writers on the local participant to be discovered, which I don't think is the desired behavior.

If what we want is that only the local participant is discovered, we could do the following:

```suggestion
      // Limit the number of participants to 1 (the local participant)
      domainParticipantQos.allocation().participants.initial = 1;
      domainParticipantQos.allocation().participants.maximum= 1;
      domainParticipantQos.allocation().participants.increment= 0;

      // Clear the list of multicast listening locators
      domainParticipantQos.wire_protocol().builtin.metatrafficMulticastLocatorList.clear();
      // Add a unicast locator to prevent creation of default multicast locator
      eprosima::fastrtps::rtps::Locator_t default_unicast_locator;
      domainParticipantQos.wire_protocol()
      .builtin.metatrafficUnicastLocatorList.push_back(default_unicast_locator);
```

Reply:
I applied your suggestion in https://github.com/ros2/rmw_fastrtps/pull/653/commits/581ab867b71f2d0cc4aa30055384cd0119e91f6f